ocs_ci/ocs/resources/storage_cluster.py

```diff
@@ -66,6 +70,29 @@ def ocs_install_verification(timeout=600, skip_osd_distribution_check=False):
     )
     log.info(f"Check if OCS operator: {ocs_csv_name} is in Succeeded phase.")
     ocs_csv.wait_for_phase(phase="Succeeded", timeout=timeout)
+    # Verify if OCS CSV has proper version.
+    csv_version = ocs_csv.data['spec']['version']
+    ocs_version = config.ENV_DATA['ocs_version']
+    log.info(
+        f"Check if OCS version: {ocs_version} matches with CSV: {csv_version}"
+    )
+    assert ocs_version in csv_version, (
+        f"OCS version: {ocs_version} mismatch with CSV version {csv_version}"
+    )
+    # Verify if OCS CSV has the same version in provided CI build.
+    ocs_registry_image = ocs_registry_image or config.DEPLOYMENT.get(
+        'ocs_registry_image'
+    )
+    if ocs_registry_image and ocs_registry_image.endswith(".ci"):
```

Comment:
I think we can get rid of this if statement so that assert is always checked, if for some reason it doesn't end with .ci, corrections can be made to catch up.

Reply:
Then you wont be able to deploy like tag like 4.2.3-rc for example and other tags. Do we want to strick users that much?

Reply:
maybe we can get rid of .ci or .rc before any check happens?

Reply:
But the version of image tag and version of CSV matches only for this specific CI tag ends wit .ci.